### PR TITLE
Start up task: Vehicle dataset comparative evaluation models  

### LINKED ISSUE #2: [Outreachy applications] Startup task: Train and test a classification model
This is a good way to get started with the environment and the problem domain. It will also provide the basis for a test case for future work. At a minimum, you should:

- load a dataset from the repo
- train a classification model from [scikit-learn](https://scikit-learn.org/stable/supervised_learning.html)
- compute an [evaluation metric](https://scikit-learn.org/stable/modules/model_evaluation.html) on a held-out test set

Feel free to include any additional steps you feel are relevant or you are interested in trying out, such as:

- basic exploratory analysis of the dataset
- data preprocessing
- hyperparameter tuning

When you start work on this task, please post a comment here indicating which dataset and model you will be working with so that other contributors can avoid duplicating your work.

### PR DESCRIPTION
**Pull request by Sidrah-Madiha**
In this version I fixed #2  and , instead of using just one model, I used comparative evaluation for choosing the best model, please see file "vehicles_dataset_classifer_v2" and related helper file "allcustommodules" for implementation, next steps will be to tune all models separately to compare or to see if we can refine data more before using model evaluation. 
model_evaluater

### CODE REVIEW

File: dev/Sidrah-Madiha/allcustommodules.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    ''' tokenize the Class column values to numeric data'''
+    factor = pd.factorize(dataset['Class'])
+    dataset.Class = factor[0]
+    definitions = factor[1]
+    print("Updated tokenize 'Class' column - first 5 values")
+    print(dataset.Class.head())
+    print("Distinct Tokens used for converting Class column to integers")
+    print(definitions)
+    return definitions
+def train_data_test_data_split(dataset):
```
Comment (mlopatka) [inline dev/Sidrah-Madiha/allcustommodules.py]: please include a docstring explaining the functionality of `train_data_test_data_split`

File: dev/Sidrah-Madiha/allcustommodules.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+#     print(X[0])
+#     print(y[0])
+#     print(X.shape)
+#     print(y.shape)
+#     print('the data attributes columns')
+#     print(X[:5,:])
+#     print('The target variable: ')
+#     print(y[:5])
+#     Split dataset into training set and test set
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2 ,random_state = 21)
```
Comment (mlopatka) [inline dev/Sidrah-Madiha/allcustommodules.py]: what is the motivation for a `test_size=0.2` split ratio?
Please motivate the selection of this parameter.
Does performance change as a function of different test/train splits?

File: dev/Sidrah-Madiha/allcustommodules.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    return X_train, X_test, y_train, y_test
+
+def train_randomforest(X_train, y_train):
+    ''' training model on train data'''
+    classifier = RandomForestClassifier(n_estimators = 10, criterion = 'entropy', random_state = 42)
```
Comment (mlopatka) [inline dev/Sidrah-Madiha/allcustommodules.py]: Please motivate the selection of these parameter.
Does performance change as a function of different parameter choices?

File: dev/Sidrah-Madiha/allcustommodules.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    classifier.fit(X_train, y_train)
+    return classifier
+
+def train_SVM(X_train, y_train):
+    params_grid = [{'kernel': ['rbf'], 'gamma': [1e-3, 1e-4],'C': [1, 10, 100, 1000]}]
```
Comment (mlopatka) [inline dev/Sidrah-Madiha/allcustommodules.py]: Please motivate the selection of these parameter.
Does performance change as a function of different parameter choices?

**Comment by dzeber** [comment]
Please remove the v1 notebook from this PR. We will merge that one from #13 separately, so they will both end up in the repo once everything is merged.

**Comment by Sidrah-Madiha** [comment]
@mlopatka this time I used the strategy of first evaluating models based on their default parameters first and based on their accuracy as well as f1 score and then improved the selected model's hyperparameter with grid-search, but I am not seeing any improvement, can you please suggest me if my process of comparison is correct, or should I tune each model first then comapre their metric? I m thinking of doing that in next iteration. Also:
- I have used black to format all .py helper files.
- I have added doc string for the function
- I have added a learning curve and table from function in train_test_split_v2.py file to motivate train
test split as well as use of grid search to motivate hyperparameter of selected model.

Please review, looking forward to your feedback.

**Comment by mlopatka** [comment]
Please squash commits before merging in to master.

**Comment by Sidrah-Madiha** [comment]
@mlopatka I need some guidance,  I am not sure what squashing commits means here, please see my last 2 commits on this branch: in second last commit I removed files in dev/Sidrah-Madiha folder that are present in master and are a part of #13 , in last commit I created a folder for this branch and added some files to it 
squash, what should be the next steps for squashing? should I force push the last commit again to my remote branch?

**Comment by dzeber** [comment]
@Sidrah-Madiha your notebook is looking great! One last thing: please change the file path to the dataset in your notebook to a relative path in the repo so that it can be run without modification.

Squashing means combining all the commits you made in this branch into a single commit. That way, when this PR is merged it looks like a single commit in the master branch history. However, I just checked and that's actually something we can do automatically when merging, so you don't need to worry about that. We will merge as soon as you've fixed the file path.

**Comment by mlopatka** [comment]
@Sidrah-Madiha Please let me know when you have addressed @dzeber 's comment and I'll merge this in.

**Comment by dzeber** [comment]
@Sidrah-Madiha I have just merged #23. Let us know if you consider this PR superceded by that one, as it satisfies #2 as well, and we can just close this one.

**Comment by Sidrah-Madiha** [comment]
@dzeber apologies for delay,  fixed relative path, branch is ready to merge
